### DISCUSSION TITLE
Recreate new tables from scratch without nuking database

### DISCUSSION THREAD

**Discussion by net-caster** [prisma/prisma#4586] [Q&A] [answered]
Hi everyone,

Before I ask my question let me give you some context:
Before the newest version (2.13.0) of Prisma I used to be able to delete the **migrations** folder in my project, delete the **_Migrations** table along with all the other tables that I wanted recreated from scratch again. This way Prisma would recreate the deleted tables without touching the other ones next time I ran the migration command.

Now if I try to do that Prisma would ask me to obliterate my db before doing anything at all.

So my question is:
How do I do that again?

Thank you

**Comment by ryands17** [comment, marked as answer]
Hey @net-caster 👋 

Yes, this is an expected change as the new migrations are deterministic. If you would want the above behaviour, you would need to use the following command:

```
prisma db push --preview-feature
```

This will create the remining tables that were deleted.

You can read more about that [here](https://www.notion.so/prismaio/Documentation-for-Prisma-Migrate-Early-Access-7708df0c1d3d4852ab39fb69ed182786#77f2f641d248437da83f53a9e1dad8d1).

Note: This command is just for development and shouldn't be used in production, so from a high level:
- `prisma db push` is typically used while prototyping and applies incremental changes to the DB  
- `prisma migrate dev` is typically used after prototyping when the DB changes are ready to be applied to other environments and when collaborating with other developers where the goal is to ensure consistent DB schemas across multiple environments.
- `prisma migrate deploy` is used after creating the migrations with `prisma migrate dev` to apply the migrations to the production database. With `deploy`, the database is never reset.

**Reply by net-caster** [reply]
Thanks, Ryan

That command does exactly what I expect it to. But what if I have a dev database that I want to convert to production and I want to keep the old data in? From what I read in the documentation and what you just said running the migrate command will always reset the database?

Thanks

**Reply by ryands17** [reply]
I would suggest using a combination of `db push` and `migrate` then.

Create your migrations with `migrate`. Those will be persisted in the DB and you can then deploy the same on production.

Use `db push` in your dev db to to drop and re-create the tables. In this case, the migrations will also remain that you can use `prisma migrate deploy --preview-feature` on your production DB to get all the migrations.

**Reply by net-caster** [reply]
Hmm, I guess this will be the new way of doing things from now on. I'll give that strategy a go.

Thanks again, Ryan

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
